src/Typescriptr/TypeScriptGenerator.cs

```diff
@@ -161,6 +169,10 @@ public GenerationResult Generate(IEnumerable<Type> types)
         private void RenderEnum(StringBuilder builder, Type enumType)
         {
             var enumString = _enumFormatter(enumType, _quoteStyle);
+            if (_typeDecorator != null)
```

Comment:
I wonder if we could do this in one place above in the while loop with something like:
`(type.IsEnum ? enumBuilder : typeBuilder).AppendLine(_typeDecorator(type));`

Little ugly duplicating the if, but it feels right keeping the decorator call in one place.